src/script/lua_api/l_localplayer.cpp

```diff
@@ -272,6 +274,71 @@ int LuaLocalPlayer::l_get_movement(lua_State *L)
 	return 1;
 }
 
+
+// hud_add(self, form)
+int LuaLocalPlayer::l_hud_add(lua_State *L)
+{
+	LocalPlayer *player = getobject(L, 1);
+
+	HudElement *elem = new HudElement;
+	read_hud_element(L, elem);
+
+	u32 id = player->addHud(elem);
+	if (id == U32_MAX) {
+		delete elem;
+		return 0;
+	}
+	lua_pushnumber(L, id);
+	return 1;
+}
+
+// hud_remove(self, id)
+int LuaLocalPlayer::l_hud_remove(lua_State *L)
+{
+	LocalPlayer *player = getobject(L, 1);
+	u32 id = luaL_checkinteger(L, 2);
+	HudElement *element = player->removeHud(id);
+	if (!element)
+		lua_pushboolean(L, false);
+	else
+		lua_pushboolean(L, true);
+	delete element;
+	return 1;
+}
+
+// hud_change(self, id, stat, data)
+int LuaLocalPlayer::l_hud_change(lua_State *L)
+{
+	LocalPlayer *player = getobject(L, 1);
+
+	u32 id = luaL_checkinteger(L, 2);
+
+	HudElement *element = player->getHud(id);
+	if (!element)
+		return 0;
+
+	void *unused;
+	read_hud_change(L, element, &unused);
+
+	lua_pushboolean(L, true);
+	return 1;
+}
+
+// hud_get(self, id)
+int LuaLocalPlayer::l_hud_get(lua_State *L)
+{
+	LocalPlayer *player = getobject(L, 1);
+
+	u32 id = lua_tonumber(L, -1);
```

Comment:
Also an issue for the SSM `l_hud_get`: This function returns `0` if it's not a valid number. Returning `0` seems reasonable in that case.

Reply:
shouldn't be an issue, in most cases returning nothing is almost the same as returning `nil` Might be easier to just change the documentation for SSM to say "return nothing if invalid"

Reply:
My point is that `0` is also a valid HUD ID - this would get the wrong ID for non-numeric parameters.

Reply:
oh sorry I didn't see what you meant there, I thought you meant `l_hud_get` by this function and the `return 0` in it which resulted in it returning nothing, when the documentation said it returns `nil` in case of invalid input.